### PR TITLE
:seedling: Improve dependency analysis workflow

### PR DESCRIPTION
**Pull request by naveensrinivasan**
- Add GitHub Action for dependency analysis
- Harden GitHub Action runner
- Run OSSF Scorecard for dependency analysis

[.github/workflows/dependency-analysis.yml]
- Add a GitHub Action for dependency analysis
- Harden the GitHub Action runner
- Checkout code for dependency analysis
- Run the dependency analysis using the OSSF Scorecard

#### What kind of change does this PR introduce?

(Is it a bug fix, feature, docs update, something else?)

- [ ] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the new behavior (if this is a feature change)?**

- [ ] Tests for the changes have been added (for bug fixes/features)

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note

```

### CODE REVIEW

File: .github/workflows/dependency-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      GITHUB_ACTOR: ${{ github.actor }}
+
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@128a63446a954579617e875aaab7d2978154e969 # v2.4.0
+        with:
+          egress-policy: audit
+
+      - name: Checkout code
+        uses: actions/checkout@ee0669bd1cc54295c223e0bb666b733df41de1c5 # v2.7.0
```
Comment (spencerschrock) [inline .github/workflows/dependency-analysis.yml:47]: Do we need to checkout anything here? My understanding is `ossf/scorecard-dependencyanalysis` just does API calls?

If we do need it, lets make sure these are up-to-date
`actions/checkout@f095bcc56b7c2baf48f3ac70d6d6782f4f553222 # v3.5.2`

File: .github/workflows/dependency-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        with:
+          persist-credentials: false
+
+      - name: Run analysis
+        uses: ossf/scorecard-dependencyanalysis@dependency-analysis
```
Comment (spencerschrock) [inline .github/workflows/dependency-analysis.yml:52]: I'm assuming it's pinned to the branch while the action is in development?
We should comeback and pin this later.

File: .github/workflows/dependency-analysis.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+on:
+  pull_request_target:
+    types:
+      - opened
+      - synchronize
+      - reopened
+
+permissions:
+  pull-requests: write
```
Comment (spencerschrock) [inline .github/workflows/dependency-analysis.yml:26]: comment on why we need the permission?

**Review by spencerschrock — APPROVED** [review]
I assume this check will not be `Required` for now?

File: .github/workflows/dependency-analysis.yml (hunk 3 above)
Comment (naveensrinivasan) [reply .github/workflows/dependency-analysis.yml:26]: We need this permission to write comments on the PR.

Comment (spencerschrock) [reply .github/workflows/dependency-analysis.yml:26]: I know, I'm asking for that to be added as a comment in the file

Comment (naveensrinivasan) [reply .github/workflows/dependency-analysis.yml:26]: OK, I will do that.

**Pull request closed without merging**